app/models/graders/junit_grader.rb

```diff
@@ -459,8 +471,13 @@ def put_job_json_with_retry!(orca_uri, job_json)
       break if attempts == max_requests
       sleep(2**attempts + rand)
     end
-    response.value
-    JSON.parse(response.body)['status']
+    if response.code.to_i != 200
+      body = JSON.parse(response.body)
+      status = { response_code: response.code.to_i }
+      return body['errors'] ? { errors: body['errors'], **status } : status
+    else
+      return JSON.parse(response.body)['status']
```

Comment:
Is this the same shape as `{ response_code: ## }` above?  Or is this something entirely different?

Reply:
This is something I'm implementing on the Bottlenose side. In essence, we either return the object from `status` on success or if we get a bad HTTP code then we return `{ response_code: int, errors?: string[] }`

Reply:
That misses my point.  I'm asking whether this method returns something at a consistent type, and if so what is that type?  Otherwise I don't know how to read the call-sites of this method to confirm that they're plausible

Reply:
This returns one of two hashes:
1. `{ 'id' => <int>, 'location' => <'Queue'|'HoldingPen'>}` or
2. `{ response_code: <int>, errors?: <string[]> }`

Reply:
Well, istm that you might want to (a) make both hashes have symbolic keys or both have string keys, but not a mix, and (b) probably always include a status field, so that if the status is 200, the id field is present, and if not, the errors field is present.

Reply:
Status is reserved for the object specified in 1) (seems I missed the full object description here), so I will attach response_code to both, but nonetheless good points